packages/atomic/src/decorators/watch.ts

```diff
@@ -0,0 +1,51 @@
+import {LitElement} from 'lit';
+
+interface WatchedPropertiesConstructor extends Function {
+  _watchedProperties?: Map<string | number | symbol, string[]>;
+}
+
+type PrivateProps = `_${string}`;
+type ProtectedProps = `#${string}`;
+type PublicProperties<T> = {
+  [K in keyof T]: K extends PrivateProps | ProtectedProps | keyof LitElement
+    ? never
+    : T[K] extends Function
+      ? never
+      : K;
+}[keyof T];
+
+// TODO: KIT-3822: add unit tests to this decorator
+export function watch<Component extends LitElement>(
```

Comment:
I think it would be usefull to add some documentation to explain in a nutshell what this decorator does, and maybe cover the implications of the `willUpdate` method override.